### PR TITLE
refactor(core): Centralize SSH Tunnel management

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR:
1. Adds SSH tunnel support for Postgres Trigger Node
1. Moves SSH client management for SSH tunnels into `n8n-core`, so that we can add tunneling support for other nodes more easily
2. Improves type-safety, and deleted redundant code in mysql and postgres nodes
3. Re-uses SSH Tunnels for upto 5 minutes, making node execution a bit faster

Not implemented in this PR:
1. Centralization of local proxies (like the one we use for the postgres node). This would help reduce code duplication even more, and would also help make nodes code simpler.

## Related Linear tickets, Github issues, and Community forum posts

- https://community.n8n.io/t/unable-to-connect-postgres-db-with-n8n/39615
- https://linear.app/n8n/issue/NODE-1267
- https://github.com/n8n-io/n8n/issues/8970

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included

### CODE REVIEW

File: packages/core/test/SSHClientsManager.test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	it('should cleanup stale SSH connections', async () => {
+		await sshClientsManager.getClient({ ...credentials, sshHost: 'host1' });
+		await sshClientsManager.getClient({ ...credentials, sshHost: 'host2' });
+		await sshClientsManager.getClient({ ...credentials, sshHost: 'host3' });
+
+		jest.advanceTimersByTime(6 * 60 * 1000);
+		sshClientsManager.cleanupStaleConnections();
+
+		expect(endSpy).toHaveBeenCalledTimes(3);
+		expect(sshClientsManager.clients.size).toBe(0);
```
Comment (despairblue) [inline packages/core/test/SSHClientsManager.test.ts:65]: image

I get a type error here. How about this:

```suggestion
		// eslint-disable-next-line @typescript-eslint/no-explicit-any
		expect((sshClientsManager as any).clients.size).toBe(0);
```

Comment (netroy) [reply packages/core/test/SSHClientsManager.test.ts:65]: interesting that the linter did not catch this 🤔

Comment (despairblue) [reply packages/core/test/SSHClientsManager.test.ts:65]: It's a TS error for me, not an eslint error.

Comment (netroy) [reply packages/core/test/SSHClientsManager.test.ts:65]: ah yes, we still don't have type-checking enabled on all backend packages. I'll fix that soon.

**Review by elsmr — COMMENTED** [review]
Nodes code looks good to me, nice one! @michael-radency could you test that there are no regressions with postgres/mysql via SSH tunnel? I think you have more experience with it

**Review by michael-radency — APPROVED** [review]
LGTM, no regressions founded, code looks good :+1:

**Comment by janober** [comment]
Got released with [`n8n@1.50.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.50.0)

**Pull request merged**